src/server/lib/queryParams.ts

```diff
@@ -71,12 +76,28 @@ export const addReturnUrlToPath = (path: string, returnUrl: string): string => {
   return `${path}${divider}returnUrl=${encodeURIComponent(returnUrl)}`;
 };
 
-export const addRefViewIdToPath = (path: string, refViewId: string) => {
-  const divider = path.includes('?') ? '&' : '?';
-  return `${path}${divider}refViewId=${encodeURIComponent(refViewId)}`;
-};
+export const getSafeQueryParams = (params: QueryParams): SafeQueryParams => ({
+  returnUrl: params.returnUrl,
+  clientId: params.clientId,
+  ref: params.ref,
+  refViewId: params.refViewId,
+});
 
-export const addRefToPath = (path: string, ref: string) => {
+export const addQueryParamsToPath = (
+  path: string,
+  params: QueryParams,
+  unsafeParams?: UnsafeQueryParams,
+): string => {
   const divider = path.includes('?') ? '&' : '?';
-  return `${path}${divider}ref=${encodeURIComponent(ref)}`;
+  const safeQueryString = stringify(getSafeQueryParams(params), {
+    skipNull: true,
+    skipEmptyString: true,
+  });
+  const unsafeQueryString = stringify(unsafeParams || {}, {
+    skipNull: true,
+    skipEmptyString: true,
+  });
+  return `${path}${divider}${safeQueryString}${
```

Comment:
is it worth using [`UrlSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) here

Reply:
Ignore me ;)